plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/ghost/TextRenderingUtils.java

```diff
@@ -141,4 +144,27 @@ public static int widgetOffset2ModelOffset(
             ? ext5.widgetOffset2ModelOffset(widgetOffset)
             : widgetOffset;
     }
+
+    public static String removeOverlap(String originalText, String suggestion) {
+        if (originalText == null || suggestion == null || originalText.isEmpty() || suggestion.isEmpty()) {
+            return suggestion;
+        }
+
+        String endOfOriginal = originalText.length() <= 20 ?
+            originalText.trim() : originalText.substring(originalText.length() - 20).trim();
+
+        endOfOriginal = endOfOriginal.replaceAll("\\s+", " ").trim().toLowerCase();
+        String cleanSuggestion = suggestion.replaceAll("\\s+", " ").trim();
+
+        for (int length = Math.min(endOfOriginal.length(), cleanSuggestion.length()); length > 0; length--) {
+            if (endOfOriginal.length() >= length) {
+                String endSubstring = endOfOriginal.substring(endOfOriginal.length() - length);
+                if (cleanSuggestion.toLowerCase().startsWith(endSubstring)) {
+                    return cleanSuggestion.substring(length).trim();
```

Comment:
We have already called trim on cleanSuggestion at line 157. Why calling it again?